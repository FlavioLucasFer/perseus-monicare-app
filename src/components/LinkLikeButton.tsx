import React from 'react';
import { 
	TouchableOpacity, 
	StyleSheet, 
	Text, 
	TouchableOpacityProps, 
} from 'react-native';

interface LinkLikeButtonProps extends TouchableOpacityProps {
	title: string,
};

const styles = StyleSheet.create({
	title: {
		color: '#0075FF',
		textDecorationLine: 'underline',
	},
});

const LinkLikeButton: React.FC<LinkLikeButtonProps> = props => {
	const {
		title,
		...rest
	} = props;

	return (
		<TouchableOpacity {...rest}>
			<Text style={styles.title}>
				{title}
			</Text>
		</TouchableOpacity>
	);
}

export type {
	LinkLikeButtonProps,
};
export default LinkLikeButton;
