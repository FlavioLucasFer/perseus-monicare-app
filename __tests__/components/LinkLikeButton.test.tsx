import React from 'react';
import { shallow } from 'enzyme';

import LinkLikeButton from 'components/LinkLikeButton';

describe('LinkLikeButton component', () => {
	it('Should render properly', () => {
		const wrapper = shallow(
			<LinkLikeButton title='teste' />
		);
		expect(wrapper.debug()).toMatchSnapshot();
	});
});
