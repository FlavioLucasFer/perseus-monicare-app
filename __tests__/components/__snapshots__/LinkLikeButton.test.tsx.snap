// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkLikeButton component Should render properly 1`] = `
"<TouchableOpacity>
  <Text style={{...}}>
    teste
  </Text>
</TouchableOpacity>"
`;
